PWGHF/D2H/TableProducer/candidateCreatorCharmResoReduced.cxx

```diff
@@ -311,8 +315,12 @@ struct HfCandidateCreatorCharmResoReduced {
         float ptReso = RecoDecay::pt(RecoDecay::sumOfVec(pVecV0Tr, pVecD));
 
         if constexpr (channel == DecayChannel::DstarTrack) {
-          invMassReso = RecoDecay::m(std::array{pVecD, pVecV0Tr}, std::array{massDstar, massProton});
-          registry.fill(HIST("hMassDstarTrack"), invMassReso, ptReso);
+          if (candD.dType() > 0) {
+            invMassReso = RecoDecay::m(std::array{candD.pVectorProng0(), candD.pVectorProng1(), candD.pVectorProng2(), pVecV0Tr}, std::array{massPion, massKaon, massPion, massProton});
+          } else {
+            invMassReso = RecoDecay::m(std::array{candD.pVectorProng1(), candD.pVectorProng0(), candD.pVectorProng2(), pVecV0Tr}, std::array{massPion, massKaon, massPion, massProton});
+          }
```

Comment:
A suggestion (not major): You are unnecessarily calling the 3 `pVector` methods twice. You can instead save the array of all 4 momenta in a single variable and reuse it for both calls of `m`, while just swapping the masses. This is not an issue in this case because we store the xyz components but for example in case of tracks, it would cause unnecessary calculations.

Reply:
Thanks Vit for the suggestion! I would implement it in the next PR, if for you it is fine!